### ISSUE TITLE
Truncation in SAC Loss

### ISSUE THREAD

**Issue by Egiob** [google/brax#172]
In SAC, the critic loss contains the following lines and I wondered why are they necessary ?
```
    # Better bootstrapping for truncated episodes.
    q_error *= jnp.expand_dims(1 - transitions.truncation_t, -1)
```
To my understanding it will zero-out the loss terms where the transitions are last transitions but only in the case where the EpisodeWrapper raises a done. Am i missing something? (and why should we do it in Brax and not in a classic Gym env ?) Thanks!

**Comment by cdfreeman-google** [comment]
(feel free to reopen if anything isn't clear!)
